Las vistas se pueden filtrar por obra, o se pueden navegar todos los modelos

fauno opened the pull request.

fauno committed:
Las obras tienen facturas

fauno committed:
Las obras tienen facturas

fauno committed:
Las obras tienen facturas

fauno committed:
las facturas pertenecen a una obra

fauno committed:
las facturas pertenecen a una obra

fauno committed:
las facturas pertenecen a una obra

fauno committed:
filtrar vistas por obras

fauno committed:
filtrar vistas por obras

fauno committed:
filtrar vistas por obras

fauno committed:
mejorcito

fauno committed:
con_obra? es un helper que decide si agregar /obra/:obra_id a las url

fauno committed:
con_obra? es un helper que decide si agregar /obra/:obra_id a las url

fauno committed:
chequear si @obra es un registro existente

fauno committed:
chequear si @obra es un registro existente

fauno committed:
decidir si filtramos por obra en todas las vistas

fauno committed:
decidir si filtramos por obra en todas las vistas

fauno committed:
Merge branch 'feature/obra-anidada' of github.com:lainventoria/cp into feature/obra-anidada

Conflicts:
	app/controllers/cajas_controller.rb

fauno committed:
mostrar la obra en el titulo

fauno committed:
mostrar la obra en el titulo

m4tu5 commented on app/controllers/cheques_controller.rb:
```diff
@@ -1,9 +1,10 @@
 
   def index
-    @cheques = Cheque.all
+    @cheques = @obra ? @obra.cheques : Cheque.all
```
@obra.cheques existe?

fauno replied:
uh ahi lo cambio por caja

m4tu5 replied:
esto no parece corregido

fauno replied:
m4tu5 notifications@github.com writes:

> >    def index
> > -    @cheques = Cheque.all
> > -    @cheques = @obra ? @obra.cheques : Cheque.all
> 
> esto no parece corregido

todavía no llegué a eso

## 

:O

m4tu5 commented:
no encuentro los cambios al controlador de recibos ... no hace falta modificarlo para poder aceder a 
obras/:obraid/recibos/pagos ?

m4tu5 commented on config/routes.rb:
```diff
@@ -1,7 +1,37 @@
+    resources :cajas
+
+    resources :facturas do
+    # Filtrar facturas por situacion
```
aca se repiten muchas definiciones que ya existen es este mismo archivo mas abajo ... debe haber alguna forma de poder aprovechar lo que ya existe sin tener que volver a tipearlo.

veamos si encontramos algo, esto es un bardo.

fauno replied:
ahi hice una limpieza para que todas las modificaciones solo se hagan con dentro de /obras/:obra_id, el tema es que ahora se rompió todo y me acordé por qué hicimos con_obra? en primer lugar :P

fauno replied:
bueno esto es un bardo...

fauno replied:
lo deje asi todo sucio porque se rompen muchas vistas, ver el commit que dice revert

m4tu5 commented on db/migrate/20140318231717_add_obra_to_factura.rb:
```diff
@@ -0,0 +1,5 @@
+class AddObraToFactura < ActiveRecord::Migration
+  def change
```
aca falta lo que hablamos de crear los datos necesarios para que los datos existentes en la bd sigan siendo validos ... en este caso, asignar una obra_id de entre los existentes de forma secuencial u aleatoria.

m4tu5 replied:
esto tambien estaria bien, asi no se nos rompe la data que ya tenemos. no quiero tener que ponerme a editar .yml a mano otra vez :)

m4tu5 commented:
estaria bien incluir algun test que pruebe lo minimo de la navegacion ... al menos, que funciona el metodo con_obra?

hecho eso, +1

fauno committed:
Paths anidados

* facturas, cajas, cheques y recibos se crean con una obra asociada
* /{facturas,cajas,cheques,recibos} son listados con distintos niveles
  de filtrado

fauno committed:
Paths anidados

* facturas, cajas, cheques y recibos se crean con una obra asociada
* /{facturas,cajas,cheques,recibos} son listados con distintos niveles
  de filtrado

fauno committed:
los cheques se filtran por caja no por obra

fauno committed:
Revert "Paths anidados"

This reverts commit a72542430e48539e2b87c2e138874896de420324.

m4tu5 committed:
diagramas de medios de pago

mauriciopasquier commented on app/controllers/cajas_controller.rb:
```diff
@@ -78,4 +79,8 @@ def set_movimientos
       params.require(:caja).permit(:obra_id, :situacion, :tipo, :banco, :numero)
     end
+
+    def set_obra
```
esto se puede mover al `ApplicationController` (debajo de `protected`, en vez de `private`), y poner el before filter nomás en cada controlador que lo vaya a usar

m4tu5 committed:
Merge branch 'feature/correccion-renderizado-parciales' into develop

fauno committed:
Merge branch 'feature/obra-anidada' of github.com:lainventoria/cp into feature/obra-anidada

Conflicts:
	app/controllers/cheques_controller.rb
	app/helpers/application_helper.rb
	config/routes.rb

fauno committed:
set_obra a applicationcontroller

fauno committed:
facturas, filtrar pagos y cobros por obra

fauno committed:
facturas: permitir pasar obra como parametro

fauno committed:
Filtrar los recibos por obra

fauno commented:
no se me ocurre como testear con_obra?

m4tu5 commented:
Podes armar algo con assert redirect?

m4tu5 commented:
Podes armar algo con assert redirect?

fauno committed:
asociar facturas con obras al azar

fauno committed:
generar un listado de obras disponibles #60

fauno committed:
yo solo obedeci ordenes

fauno committed:
generar un menu de obra para todas las vistas  #60

el problema es que asi como estan las rutas, /obra/1/factura/1 y
/obra/2/factura/1 siempre devuelven factura/1 sin fijarse en la obra

fauno committed:
redirigir al indice si estamos en un show #60

fauno committed:
generar el listado de obras correctamente :) close #60

m4tu5 commented:
+1
